tests/test_timechop.py

```diff
@@ -188,27 +188,90 @@ def test_valid_input(self):
+                        datetime.datetime(2010, 1, 12, 0, 0),
+                        datetime.datetime(2010, 1, 13, 0, 0),
+                        datetime.datetime(2010, 1, 14, 0, 0),
+                        datetime.datetime(2010, 1, 15, 0, 0)
+                    ]
+                }]
+            }
+        ]
+
         chopper = Inspections(
             beginning_of_time = datetime.datetime(1990, 1, 1, 0, 0),
             modeling_start_time = datetime.datetime(2010, 1, 1, 0, 0),
             modeling_end_time = datetime.datetime(2010, 1, 16, 0, 0),
             update_window = '5 days',
-            look_back_durations = ['6 days']
+            look_back_durations = ['7 days']
         )
-        with self.assertRaises(ValueError):
-            chopper.chop_time()
+        result = chopper.chop_time()
+        assert(result == expected_result)
+
+    def test_unevenly_divisible_update_window(self):
+        expected_result = [
+            {
+                'beginning_of_time': datetime.datetime(1990, 1, 1, 0, 0),
+                'modeling_start_time': datetime.datetime(2010, 1, 1, 0, 0),
+                'modeling_end_time': datetime.datetime(2010, 1, 16, 0, 0),
+                'train_matrix': {
+                    'matrix_start_time': datetime.datetime(2010, 1, 5, 0, 0),
+                    'matrix_end_time': datetime.datetime(2010, 1, 10, 0, 0),
+                    'as_of_times': [
+                        datetime.datetime(2010, 1, 5, 0, 0),
+                        datetime.datetime(2010, 1, 6, 0, 0),
+                        datetime.datetime(2010, 1, 7, 0, 0),
+                        datetime.datetime(2010, 1, 8, 0, 0),
+                        datetime.datetime(2010, 1, 9, 0, 0)
+                    ]
+                },
+                'test_matrices': [{
+                    'matrix_start_time': datetime.datetime(2010, 1, 10, 0, 0),
+                    'matrix_end_time': datetime.datetime(2010, 1, 16, 0, 0),
+                    'as_of_times': [
+                        datetime.datetime(2010, 1, 10, 0, 0),
+                        datetime.datetime(2010, 1, 11, 0, 0),
+                        datetime.datetime(2010, 1, 12, 0, 0),
+                        datetime.datetime(2010, 1, 13, 0, 0),
+                        datetime.datetime(2010, 1, 14, 0, 0),
+                        datetime.datetime(2010, 1, 15, 0, 0)
+                    ]
+                }]
+            }
+        ]
 
-    def test_bad_update_window(self):
         chopper = Inspections(
             beginning_of_time = datetime.datetime(1990, 1, 1, 0, 0),
             modeling_start_time = datetime.datetime(2010, 1, 1, 0, 0),
             modeling_end_time = datetime.datetime(2010, 1, 16, 0, 0),
             update_window = '6 days',
             look_back_durations = ['5 days']
         )
-        with self.assertRaises(ValueError):
-            chopper.chop_time()
+        result = chopper.chop_time()
```

Comment:
Just pushed an update that does this. Turns out there was an error in the warning that this prompted me to resolve!